playground/environment-react-ssr/vite.config.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  type Connect,
+  type Plugin,
+  type PluginOption,
+  createServerModuleRunner,
+  defineConfig,
+} from 'vite'
+
+export default defineConfig((env) => ({
+  clearScreen: false,
+  appType: 'custom',
+  plugins: [
+    vitePluginSsrMiddleware({
+      entry: '/src/entry-server',
+      preview: new URL('./dist/server/index.js', import.meta.url).toString(),
+    }),
+    {
+      name: 'global-server',
+      configureServer(server) {
+        Object.assign(globalThis, { __globalServer: server })
+      },
+    },
+  ],
+  environments: {
+    client: {
+      build: {
+        minify: false,
+        sourcemap: true,
+        outDir: 'dist/client',
+      },
+    },
+    ssr: {
+      build: {
+        outDir: 'dist/server',
+        // [feedback]
+        // is this still meant to be used?
+        // for example, `ssr: true` seems to make `minify: false` automatically
+        // and also externalization.
+        ssr: true,
```

Comment:
The current idea is to replace `ssr` and `ssrXXX` options for fine grained options. These new options could have by default the equivalent value to the current `ssr` if we think that it is more common to create SSR like environments compared to client like environments. See for example `environment.nodeCompatible` to deal with node-builtins, that was previously checking the `ssr` flag. That still should probably be even more fine-grained.

Reply:
> The current idea is to replace ssr and ssrXXX options for fine grained options.

The idea sounds good to me. Yeah, all-or-nothing `ssr` flag doesn't seem to match with eventual goal of environment system.

I just wanted to note that the current behavior of builtin ssr build might be surprising without explicit `ssr: true` since it bundles all externals.

Reply:
Maybe we have a bug? Is the current `external: []` the wrong default? I may have introduced it because so far `srr.external` could remain as `undefined` and I'm now forcing it to be `Array | true`

Reply:
I haven't dig into this, so I'm not sure if it's a bug.

You can check the difference by toggling `ssr: true` in this playground. I assumed this is because it's bundling like a client when there's no `ssr: true`.

```sh
# with ssr: true
$ pnpm -C playground/environment-react-ssr build
...
vite v6.0.0-alpha.0 building SSR bundle for production...
✓ 3 modules transformed.
dist/server/assets/index-BILxvZ2l.js  0.40 kB
dist/server/index.js                  1.18 kB

# without ssr: true
$ pnpm -C playground/environment-react-ssr build
...

vite v6.0.0-alpha.0 building SSR bundle for production...
✓ 30 modules transformed.
dist/server/assets/index-CfOAxdGB.js    0.37 kB │ gzip:  0.25 kB
dist/server/index.js                  306.46 kB │ gzip: 96.19 kB
```